lib/rust/code-builder/src/lib.rs

```diff
@@ -122,7 +126,7 @@ impl AddToBuilder<&str> for CodeBuilder {
 
 impl AddToBuilder<String> for CodeBuilder {
     fn add_to_builder(&mut self, t: String) -> &mut Self {
-        self.add_str(t);
+        self.add_str(&t);
         self
```

Comment:
if this is changed, then why `add_to_builder` consumes `String`?

Reply:
This is a trait implementation. That specific impl is on `String` in this case, there are other for refs next to it. In my opinion we shouldn't be implementing this on string, as it potentially hides unnecessary allocations by consuming those strings. But I didn't wanted to change too much outside of this file.

Reply:
I removed the String implementation of this trait. The only usage of it was for usize, which is better handled by using `write_fmt`, as it doesn't allocate temporary string.